docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+Here we download it and reference it locally.
+Note that we get this in `.tar.gz` format, so you need to decompress this before using the data.
+Once you decompress, note the location as a constant in the `constants.py` module we created earlier.
+Something like this will do:
+
+`DATA_SRC = '/path/to/archive/aclImdb_v1.tar/'`
+
+Next, open up `backend/ayx_plugins/text_classifier.py`. It's finally time to add some functional code!
+
+Update the `__init__` method to the below:
+
+```python
+    def __init__(self, provider: "AMPProviderV2") -> None:
+        """Construct a plugin."""
+        self.provider = provider
+
+        self.MODE = "DATA"
+
+        # The location of the raw data
+        self.data_url = self.provider.tool_config["datasetConfig"]["datasetTargetDir"]
+        # The location to store training datasets for our model
+        self.train_dir = self.provider.tool_config["datasetConfig"]["trainingSetDir"]
+        # Test datasets, like the training above.
+        self.test_dir = self.provider.tool_config["datasetConfig"]["testSetDir"]
+        # We pull validation datasets from the same location as our training here
+        self.val_dir = self.train_dir
+
+        self.provider.io.info("Plugin initialized.")
+```
+
+Where:
+
+`self.MODE = "DATA"`
+ is our tool's "run mode," as described earlier in [link to section].
+
+And:
+
+ `self.data_url = self.provider.tool_config["datasetConfig"]["datasetTargetDir"]` 
+
+ And the lines below it utilize the `tool_config` dict (link to tool_config in docs) to access data and tool configuration sent by the UI SDK.
+
+You may have already noticed, but `tool_config` will contain the values we said we'd mock before deploying to Designer.
+Once we have developed our frontend, a user can populate these fields.
+Until then, you can safely use constants for testing. (better phrasing?)
+
+Next, add the following to the `on_complete` method:
+
+```python
+try:
+  if self.MODE == 'DATA':
+    self.setup_data()
+except Exception as e:
+  logger = logger.error(f"{self.on_complete}: repr(e)")
+  self.provider.io.error(f"{self.on_complete}: an error occuried, please see PythonSDK.log for details!")
+```
+#### Supplemental: Error Handling in the SDK
+
+There are two crucial pieces to note here. If you already understand the concepts below, feel free to skip to the next section!
+
+First, the first Python sdk tool runs as a separate (python) process from Designer's process.
+While this provides many benefits (such as arbitrary Python tool requirements like TensorFlow here), this means *designer* cannot directly access the exceptions the Python process throws.
+The SDK does its best to capture generic errors, but the scope of edge cases is impossibly large. 
+It is possible to try and catch the "wrong" error type in Python and unintentionally let an exception pass silently. (See Python docs for details)
+As such, we have seen user cases where tricky exceptions slip off the stack uncaught.
```

Comment:
TW review:

There are 2 crucial pieces to note here. If you already understand the concepts below, feel free to skip to the next section.

First, the first Python SDK tool runs as a separate (python) process from Designer's process. While this provides many benefits (such as arbitrary Python tool requirements like TensorFlow here), this means Designer can't directly access the exceptions the Python process throws. The SDK does its best to capture generic errors, but the scope of edge cases is impossibly large. It's possible to try and catch the wrong error type in Python and unintentionally let an exception pass silently. As such, we have seen use cases where tricky exceptions slip off the stack uncaught.